code-collab-app/src/storage/peer_data.js

```diff
@@ -0,0 +1,142 @@
+import Peer from 'peerjs';
+import {observable, action, computed} from 'mobx';
+import io from 'socket.io-client';
+
+const peer_data = observable({
+    //  hard coded data for now
+    session_list: ['test1', 'test2', 'test3'],
+    session_peers: [],
+    session_peers_conn: [],
+    peer: null,
+
+    peer_id: -1,
+    current_session: 'test1',
+    current_session_id: '1',                        //  in case 2 files have the same name
+    socket: null,
+    doc_data: 'Hello World.',
+
+    initialize(){
+        console.log('in initialize function');
+        // will change later, hard coded for now
+        this.socket = io('http://localhost:8000/');
+        this.socket.on('session_list', function (list) {
+            this.session_list = JSON.parse(list);
+        });
+    },
+
+    create_new_session(session_name){
+        this.peer = new Peer();
+
+        this.peer.on('open', function(id){
+            this.peer_id = id;
+            console.log("id: " + this.id);
+
+            let peer_info = {   "session_name": session_name, 
+                                "peer_id": this.peer_id};
+
+            //  inform tracker new session info
+            this.socket.emit('new session', JSON.stringify(peer_info));
+
+            this.socket.on('session created', function(session_info){
+                let temp = JSON.parse(session_info);
+                this.current_session = temp.session_name;
+                this.current_session_id = temp.session_id;
+            });
+
+            this.listen_for_req();
+        })
+    },
+
+    //  Join an active session or open preexisted doc
+    join_session(session_name){
+        this.current_session = session_name;
+
+        this.peer = new Peer();
+
+        this.peer.on('open', function(id){
+            this.peer_id = id;
+            console.log("id: " + this.id);
+
+            let peer_info = {   "session_name": session_name, 
+                                "peer_id": this.peer_id};
+
+            //  Request list of peers in <session_name> session
+            this.socket.emit('join session', JSON.stringify(peer_info));
```

Comment:
With `socket.io` I think we can send objects directly without needing to `stringify` or `parse` them on the other side.